src/DotVVM.Framework/Binding/Expressions/CommandBindingExpression.cs

```diff
@@ -49,8 +49,11 @@ public class OptionsAttribute : BindingCompilationOptionsAttribute
 
             public class Methods
             {
-                public CommandJavascriptBindingProperty CreateJs(IdBindingProperty id) =>
-                    new CommandJavascriptBindingProperty(CreateJsPostbackInvocation(id.Id));
+                public CommandJavascriptBindingProperty CreateJs(IdBindingProperty id, ExpectedTypeBindingProperty expectedType = null) =>
+                    new CommandJavascriptBindingProperty(CreateJsPostbackInvocation(
+                        id.Id,
+                        needsCommandArgs: expectedType?.Type?.GetDelegateArguments()?.Length != 0
```

Comment:
It won't work when `GetDelegateArguments()?.Length` returns `null`.

Reply:
It will - when it's null, it's not equal to `0`, so command args are required.

Reply:
Ok. Then I don't get what `needsCommandArgs` really means.

Reply:
The point is to only include the commandArgs when they are required. But I have changed implementation so it also supports "strict mode" when it will throw an exception unless the commandArgs are specified.